pyrenew_hew/site_level_dynamics_model.py

```diff
@@ -0,0 +1,566 @@
+            )
+            sigma_initial_exp_growth_rate = (
+                self.sigma_initial_exp_growth_rate_rv()
+            )
+            initial_exp_growth_rate_non_ref_subpop_rv = TransformedVariable(
+                "clipped_initial_exp_growth_rate_non_ref_subpop",
+                DistributionalVariable(
+                    "initial_exp_growth_rate_non_ref_subpop_raw",
+                    dist.Normal(
+                        mean_initial_exp_growth_rate,
+                        sigma_initial_exp_growth_rate,
+                    ),
+                    reparam=LocScaleReparam(0),
+                ),
+                transforms=lambda x: jnp.clip(x, -0.01, 0.01),
+            )
+
+            autoreg_rt_subpop = self.autoreg_rt_subpop_rv()
+            sigma_rt = self.sigma_rt_rv()
+            rtu_subpop_ar_init_rv = DistributionalVariable(
+                "rtu_subpop_ar_init",
+                dist.Normal(
+                    0,
+                    sigma_rt / jnp.sqrt(1 - jnp.pow(autoreg_rt_subpop, 2)),
+                ),
+            )
+
+            with numpyro.plate("n_subpops", self.n_subpops - 1):
+                initial_exp_growth_rate_non_ref_subpop = (
+                    initial_exp_growth_rate_non_ref_subpop_rv()
+                )
+                i_first_obs_over_n_non_ref_subpop = (
+                    i_first_obs_over_n_non_ref_subpop_rv()
+                )
+                rtu_subpop_ar_init = rtu_subpop_ar_init_rv()
+
+            i_first_obs_over_n_subpop = jnp.hstack(
+                [
+                    i_first_obs_over_n_ref_subpop,
+                    i_first_obs_over_n_non_ref_subpop,
+                ]
+            )
+            initial_exp_growth_rate_subpop = jnp.hstack(
+                [
+                    initial_exp_growth_rate_ref_subpop,
+                    initial_exp_growth_rate_non_ref_subpop,
+                ]
+            )
+
+            rtu_subpop_ar_proc = ARProcess()
+            rtu_subpop_ar_weekly = rtu_subpop_ar_proc(
+                noise_name="rtu_ar_proc",
+                n=n_weeks_post_init,
+                init_vals=rtu_subpop_ar_init[jnp.newaxis],
+                autoreg=autoreg_rt_subpop[jnp.newaxis],
+                noise_sd=sigma_rt,
+            )
+            numpyro.deterministic("rtu_subpop_ar_weekly", rtu_subpop_ar_weekly)
+            log_rtu_non_ref_subpop_in_week = (
+                rtu_subpop_ar_weekly + log_r_t_in_weeks[:, jnp.newaxis]
+            )
+            log_rtu_subpop_in_week = jnp.concat(
+                [
+                    log_rtu_ref_subpop_in_week[:, jnp.newaxis],
+                    log_rtu_non_ref_subpop_in_week,
+                ],
+                axis=1,
+            )
+        else:
+            i_first_obs_over_n_subpop = i_first_obs_over_n_ref_subpop
+            initial_exp_growth_rate_subpop = initial_exp_growth_rate_ref_subpop
+            log_rtu_subpop_in_week = log_rtu_ref_subpop_in_week[:, jnp.newaxis]
```

Comment:
Mild preference for reversing the order of the if else here. (If <= 1 subpop, do the simple thing. Else do the complex thing) By the time I got to this, I forgot what the `if` was checking.

Reply:
Good point! changed in https://github.com/CDCgov/pyrenew-hew/pull/49/commits/259e895ce55a0248dca4ffd590dc418dfb172dc9